libguard/guard_interface.cpp

```diff
@@ -260,14 +260,19 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType,
     return getHostEndiannessRecord(guard);
 }
 
-GuardRecords getAll()
+GuardRecords getAll(bool persistentTypeOnly)
```

Comment:
The direction used here is get all the available gurad records using this api, It is better to implment lib users to filterout the require guard records here. like guard tool  implemented  https://github.com/open-power/guard/blob/main/guard.cpp#L27  .

Reply:
Still, this API supports to return of all records i.e by default it returns all records.

IMO, we are forcing the caller to do the filter that he doesn't want to do forever as per the direction and one more issue that I am seeing is, whenever introducing a new ephemeral type of record, we are expecting to change in the caller side too so, this change allows us to keep all in the one place and also give the flexibility to get all records too.

are you seeing any problem with this proposal?

Reply:
The commit i mentioned earlier is also pushed by you and i remember we were discussion same  and decided to go with the direction i mentioned earlier. .Better be consistent in guard tool inxcase you want to go with api change unless ther is strong reason to do that.

Reply:
ah ok, I did not push that code, it is already there before I start to contribute to the guard. I have extended that for a new type.

Now, I am trying to fix something else but, due to this filter logic at the caller end, it forced me to do extra computation that's not needed by the caller forever. 

I can change the guard tool code in the same way.